### ISSUE TITLE
Google Cast No Longer Works

### ISSUE THREAD

**Issue by v4w5gse65g874** [jellyfin/jellyfin-android#1501]
### Describe the bug

After upgrading the webserver via docker to 10.10.0, and clearing the App Cache of Jellyfin android app (login wouldn't work otherwise) the Google Cast button in the app, when clicked, just spins and times out whereas on the previous version it displayed devices to cast to.

### Logs

```shell
NA
```

### Application version

2.6.2

### Where did you install the app from?

Google Play

### Device information

Google Pixel 4a

### Android version

Android 13

### Jellyfin server version

10.10.0

### Which video player implementations does this bug apply to?

- [X] Web player (default)
- [ ] Integrated player (ExoPlayer)
- [ ] External player (VLC, mpv, MX Player)

**Comment by v4w5gse65g874** [comment]
Reviewing my server logs when clicking the cast button reveals (did 10.10.0 change the Sessions structure?):

```
[2024-10-29 03:33:09.484 +00:00] [ERR] [35] Jellyfin.Api.Middleware.ExceptionMiddleware: Error processing request. URL "GET" "/Sessions".
System.NullReferenceException: Object reference not set to an instance of an object.
   at Emby.Server.Implementations.Session.SessionManager.<>c.<GetSessions>b__110_2(SessionInfo r)
   at System.Linq.Utilities.<>c__DisplayClass2_0`3.<CombineSelectors>b__0(TSource x)
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.ToList()
   at Emby.Server.Implementations.Session.SessionManager.GetSessions(Guid userId, String deviceId, Nullable`1 activeWithinSeconds, Nullable`1 controllableUserToCheck, Boolean isApiKey)
   at Jellyfin.Api.Controllers.SessionController.GetSessions(Nullable`1 controllableByUserId, String deviceId, Nullable`1 activeWithinSeconds)
   at lambda_method1112(Closure, Object, Object[])
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeActionMethodAsync()
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeNextActionFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|25_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Jellyfin.Api.Middleware.ServerStartupMessageMiddleware.Invoke(HttpContext httpContext, IServerApplicationHost serverApplicationHost, ILocalizationManager localizationManager)
   at Jellyfin.Api.Middleware.WebSocketHandlerMiddleware.Invoke(HttpContext httpContext, IWebSocketManager webSocketManager)
   at Jellyfin.Api.Middleware.IPBasedAccessValidationMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager)
   at Jellyfin.Api.Middleware.LanFilteringMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager, IServerConfigurationManager serverConfigurationManager)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Jellyfin.Api.Middleware.QueryStringDecodingMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.ReDoc.ReDocMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Jellyfin.Api.Middleware.RobotsRedirectionMiddleware.Invoke(HttpContext httpContext)
   at Jellyfin.Api.Middleware.LegacyEmbyRouteRewriteMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.ResponseCompression.ResponseCompressionMiddleware.InvokeCore(HttpContext context)
   at Jellyfin.Api.Middleware.ResponseTimeMiddleware.Invoke(HttpContext context, IServerConfigurationManager serverConfigurationManager)
   at Jellyfin.Api.Middleware.ExceptionMiddleware.Invoke(HttpContext context)
```

**Comment by v4w5gse65g874** [comment]
It seems to me that this commit in 10.10.0 made a change to how the User object is loaded to handle API keys:

https://github.com/jellyfin/jellyfin/pull/12696/commits/5ead283731ef65a90f70115e754e6af8f5590ee1

and perhaps the Android app is not sending the user's Guid to the Session API endpoint?

**Comment by Vaear** [comment]
Sadly i am experiencing the same issue using the official android app, trying to click the "Cast" button results in a never ending "loading spin".

Hopefully this gets looked at and fixed fast, considering a downgrade to 10.9 so i am able to use my server again.

**Comment by DarkNeroXIII** [comment]
Thanks! I do have the exact same issue right now. At first i suspected a newly installed Plugin but trying around made clear it was just the upgrade to 10.10. (I do run Jellyfin directly in an lxc container without docker)

So far i did not find any solution sorry, i am trying some things

**Comment by Maxr1998** [comment]
Definitely possible that the Android app isn't sending compatible data currently, but this also seems like a server issue since it really shouldn't log this kind of exception. I'll have to look at the logs of the Android app if those contain more actionable info for us.

**Comment by Trellmor** [comment]
See jellyfin/jellyfin#12717 You should be able to reproduce it with an non-admin user.

**Comment by nielsvanvelzen** [comment]
This issue is already fixed for 10.10.1, closing as duplicate of jellyfin/jellyfin#12717.

**Comment by ispasov-hub** [comment]
Hi! Thank you for the great product! 

I am using docker to run version 10.10.1 and still not able to cast to chromecast from the android app (2.6.2). 
The playback just locks in place.
